lib/_tls_wrap.js

```diff
@@ -285,6 +285,12 @@ function TLSSocket(socket, options) {
     this._tlsOptions = {};
   else
     this._tlsOptions = options;
+
+  if (this._tlsOptions.NPNProtocols)
+    tls.convertNPNProtocols(this._tlsOptions.NPNProtocols, this._tlsOptions);
+  if (this._tlsOptions.ALPNProtocols)
+    tls.convertALPNProtocols(this._tlsOptions.ALPNProtocols, this._tlsOptions);
```

Comment:
You should be able to remove the calls from downstream functions now.

As well, can you make it so (and add a test to that effect) that the user's option object isn't modified?  Thanks.

Reply:
Yup, that had been bothering me too.